### PR TITLE
WIP french dataset

### PR DESCRIPTION
**Pull request by pchampio**
LM done

## What does this PR do?

Add recipe for common french elra dataset

<details>
  <summary><b>Before submitting</b></summary>

- [x] Did you read the [contributor guideline](https://speechbrain.readthedocs.io/en/latest/contributing.html)?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing [tests](https://github.com/speechbrain/speechbrain/tree/develop/tests) pass** locally with your changes?
- [ ] Does your code adhere to project-specific code style and conventions?

</details>

## PR review

<details>
  <summary>Reviewer checklist</summary>

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified
- [ ] Confirm that the changes adhere to compatibility requirements (e.g., Python version, platform)
- [ ] Review the self-review checklist to ensure the code is ready for review

</details>

### CODE REVIEW

**Comment by asumagic** [comment]
Squashed the commit into one so that the removed pretrained HF model stuff doesn't make it to the history; looking into the other CI issues (which seems to be because of old yaml that breaks due to the changes in the train scripts) and will try to test soon

**Comment by pchampio** [comment]
@asumagic @Adel-Moumen PR ready for review.
The dataset is private so I'm not providing the model weights.

**Comment by asumagic** [comment]
We will both be somewhat unavailable/busy in the near future but I will try to review by the end of this month.

File: recipes/ESTER+EPAC+ETAPE+REPERE/stm_prepare.py [hunk 1]
```diff
@@ -0,0 +1,615 @@
...
+    wav_directory,
+    tr_splits,
+    dev_splits,
+    te_splits,
+    save_folder,
+    merge_train_csv,
+    train_csv,
+    skip_prep=False,
+    ignore_wav=False,
+    new_word_on_apostrophe=True,
```
Comment (asumagic) [inline recipes/ESTER+EPAC+ETAPE+REPERE/stm_prepare.py:51]: Missing documentation for both

**Review by asumagic — CHANGES_REQUESTED** [review]
Reviewed as much as I could and have some minor changes suggested. If you are unavailable I will look into fixing those changes myself.

Note that I don't actually have those original datasets on hand at the work cluster so I haven't really been able to repro. As far as I'm aware, down the line the informal stance of the project is that it is OK to have a non-public dataset recipe, as long as someone is there to maintain it when there are issues. i.e. if it eventually accumulates issues without any maintainer able to fix it, it may be dropped.

Also, if it is possible and/or meaningful to document how to train the model without a LM, that would be good, as we would prefer to keep mandatory externally invoked software for recipes to a minimum.

I tested `LibriSpeech/ASR/CTC/train_with_wav2vec_k2.py` to make sure it didn't regress, though.

**Review by TParcollet — CHANGES_REQUESTED** [review]
Thank you so much for the PR @pchampio, my review is bellow. Overall it looks good, but a few things need to be double checked with the other maintainers. @asumagic the recipe **MUST** be tested from scratch before merging as it is fairly complex.

File: recipes/ESTER+EPAC+ETAPE+REPERE/ASR/README.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+ - ESTER2: https://catalogue.elra.info/en-us/repository/browse/ELRA-S0338
+ - ETAPE:  https://catalogue.elra.info/en-us/repository/browse/ELRA-E0046
+ - EPAC:   https://catalogue.elra.info/en-us/repository/browse/ELRA-S0305
+ - REPERE: https://catalogue.elra.info/en-us/repository/browse/ELRA-E0044
+
+**Supported pre-trained wav2vec2 from LeBenchmark:** [HuggingFace](https://huggingface.co/LeBenchmark)
+
+## Installing Extra Dependencies
+
+Before proceeding, ensure you have installed the necessary additional dependencies. To do this, simply run the following command in your terminal:
```
[the remark this answers is not in the document]
Comment (asumagic) [reply recipes/ESTER+EPAC+ETAPE+REPERE/ASR/README.md:14]: IME k2 does kind of need to be installed manually to select the proper CUDA version, etc. (and I think the wheel on PyPI is not updated at the same frequency), so it's better to add it there in the README.

File: recipes/ESTER+EPAC+ETAPE+REPERE/LM/README.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+pip install -r extra_requirements.txt
+```
+
+If you want to train an n-gram, you will first need to install `k2`. The integration has been tested with `k2==1.24.4` and `torch==2.0.1`, although it should also work with any `torch` version as long as `k2` supports it (compatibility list [here](https://k2-fsa.github.io/k2/installation/pre-compiled-cuda-wheels-linux/index.html)). You can install `k2` by following the instructions [here](https://k2-fsa.github.io/k2/installation/from_wheels.html#linux-cuda-example).
+If you want to train an n-gram, in this recipe we are using the popular KenLM library. Let's start by installing the Ubuntu library prerequisites. For a complete guide on how to install required dependencies, please refer to [this](https://kheafield.com/code/kenlm/dependencies/) link:
+ ```
+ sudo apt install build-essential cmake libboost-system-dev libboost-thread-dev libboost-program-options-dev libboost-test-dev libeigen3-dev zlib1g-dev libbz2-dev liblzma-dev
+ ```
+
+ Next, we need to start downloading and unpacking the KenLM repo.
```
[the remark this answers is not in the document]
Comment (asumagic) [reply recipes/ESTER+EPAC+ETAPE+REPERE/LM/README.md:25]: @pchampio told me that it isn't IIRC, but I have not actually verified personally.

**Comment by asumagic** [comment]
pre-commit fail resolution depends on #2665 merge

**Comment by asumagic** [comment]
I'll get to continuing changes when I have news about getting EPAC/REPERE in their original format for reproducing this.

**Comment by asumagic** [comment]
@pchampio could you send me the file list/hierarchy of those datasets? I have had discussions at the lab and there is some confusion about how e.g. REPERE has been obtained here (in different parts over years), so getting the "original format" might be challenging.

If we are not the only ones with this kind of trouble, it might be problematic.. I do hope we can get this sorted out but if this is too complicated it might have to stay out-of-tree IMO, although we can review and merge any change it requires even if not needed for other in-tree recipes.

**Comment by pchampio** [comment]
I'll answer to all of the comments when I get back to the office next week.

**Comment by asumagic** [comment]
No problem, ty :)

**Comment by pchampio** [comment]
I've checked the E0044.zip from `https://catalog.elra.info/en-us/repository/browse/ELRA-E0044/` and yes, in fact the file/directory hierarchy differs from the one of the zip.
The REPERE hierarchy is the following:
```
$~ ls
dev2013  dev2014   test2013   test2014   tools  train
```
In my script, I assume that the wav and `stm` (missing from the zip has to be generated with trs2stm.pl) are in the same directory.
